unlock-app/src/utils/validators.js

```diff
@@ -19,3 +27,130 @@ export const isPositiveNumber = val => {
   const parsedFloat = parseFloat(val)
   return !isNaN(parsedFloat) && +parsedFloat >= 0
 }
+
+export const isAccountOrNull = val => {
+  return val === null || (typeof val === 'string' && val.match(ACCOUNT_REGEXP))
+}
+
+export const isAccount = val => {
+  return val && typeof val === 'string' && val.match(ACCOUNT_REGEXP)
+}
+
+/**
+ * helper function to assert on a thing being an
+ * object with required and optional keys
+ *
+ * No assertion on the types of the values is done here
+ */
+function isValidObject(obj, validKeys) {
+  if (!obj || typeof obj !== 'object') return false
+  const keys = Object.keys(obj)
+
+  if (keys.length < validKeys.length) return false
+  if (validKeys.filter(key => !keys.includes(key)).length) return false
+  return true
+}
+
+/**
+ * validate a single key. This should match the type in unlockTypes.ts
+ */
+export const isValidKey = key => {
```

Comment:
There is a lot going on in this method, lets see if we can break this down a bit, totally fine if we extract some supporting methods here